src/uzfs_prop.c

```diff
@@ -0,0 +1,41 @@
+#include <zrepl_mgmt.h>
+#include <uzfs_mgmt.h>
+
+/*
+ * This file defines function to set property of uzfs zvol
+ */
+int
+uzfs_zfs_set_prop(const char *name, zprop_source_t source, nvlist_t *list)
+{
+	zvol_info_t *zinfo;
+	int error;
+
+	/* fetch zinfo for given volume */
+	zinfo = uzfs_zinfo_lookup(name);
+	if (zinfo == NULL) {
+		LOG_ERR("Invalid volume %s", name);
+		return (EINVAL);
+	}
+
+	(void) pthread_mutex_lock(&zinfo->zinfo_mutex);
```

Comment:
why do we need this lock?

Reply:
we can skip it since refcount is taking care of zv deletion part. Done.